frappe/public/js/frappe/form/formatters.js

```diff
@@ -53,6 +53,14 @@ frappe.form.formatters = {
 	Currency: function(value, docfield, options, doc) {
 		var currency = frappe.meta.get_field_currency(docfield, doc);
 		var precision = docfield.precision || cint(frappe.boot.sysdefaults.currency_precision) || 2;
+		var temp_value = cstr(value).split(".");
```

Comment:
it is bad practice to use variable names like `temp_value`. It makes the code unreadable.
```
if (precision > 2) {
  let parts = cstr(value).split('.');
  let decimals = parts.length > 0 ? parts[1] : '';
  if (decimals.length < 3) {
    // force 2 decimals irrespective of currency precision if there are less than 3 decimals
    precision = 2;
  }
}
```